src/serlo-editor-repo/plugin-text/components/text-editor.tsx

```diff
@@ -192,6 +188,26 @@ export function TextEditor(props: TextEditorProps) {
         }
       }
 
+      // Special handler for links. When pressing Enter while editing a link-text
+      // the link should not continue in the next line. Instead a new line with normal text is created.
+      if (event.key === 'Enter') {
+        const { path, offset } = selection.focus
+        const node = Node.get(editor, path)
+        const parent = Node.parent(editor, path)
+
+        if (node && parent) {
+          if (
+            Object.hasOwn(parent, 'type') &&
+            parent.type === 'a' &&
+            Object.hasOwn(node, 'text') &&
+            node.text.length === offset
+          ) {
+            Transforms.move(editor)
+            event.preventDefault()
+          }
+        }
+      }
```

Comment:
For me, this doesn't work like the comment describes it. The link will be split, and the new line will also have a link. If I didn't understand it well, maybe just the comment could be a bit more specific about the behavior.

Reply:
Nice catch, thank you. I forgot to handle all cases. Could you the behaviour again if it works now and the comments are clear?

Reply:
It's clear now and works as described, thanks for updating!